packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx

```diff
@@ -128,6 +129,43 @@ const DatagridRow: FC<DatagridRowProps> = React.forwardRef((props, ref) => {
             rowClick,
         ]
     );
+    const handleAuxClick = useCallback(
+        async event => {
+            if (!rowAuxClick) return;
+            event.preventDefault();
+
+            const effect =
+                typeof rowAuxClick === 'function'
+                    ? await rowAuxClick(id, basePath, record)
+                    : rowAuxClick;
+            switch (effect) {
+                case 'edit':
+                    history.push(linkToRecord(basePath, id));
```

Comment:
the middle click is supposed to open the link in a new tab. I don't see that in your PR.